app/middleware/authentication.rb

```diff
@@ -23,6 +23,13 @@ def call(env)
       end
 
       if user
+        if email
```

Comment:
This is a good spot to extract a well-named method to describe the operation that's happening. We could pass some of these values in as arguments and return if any are nil instead of doing a nested conditional. You can also do something like `if user.present? && email.present?` instead of nesting.